eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/AbstractMutableList.java

```diff
@@ -355,13 +356,41 @@ public T detect(Predicate<? super T> predicate)
     }
 
     @Override
+    public <P> T detectWith(Predicate2<? super T, ? super P> predicate, P parameter)
+    {
+        return ListIterate.detectWith(this, predicate, parameter);
+    }
+
+    @Override
+    public Optional<T> detectOptional(Predicate<? super T> predicate)
+    {
+        return ListIterate.detectOptional(this, predicate);
+    }
+
+    @Override
+    public <P> Optional<T> detectWithOptional(Predicate2<? super T, ? super P> predicate, P parameter)
+    {
+        return ListIterate.detectWithOptional(this, predicate, parameter);
+    }
+
+    @Override
     public T detectIfNone(Predicate<? super T> predicate, Function0<? extends T> function)
     {
         T result = this.detect(predicate);
         return result == null ? function.value() : result;
     }
 
     @Override
+    public <P> T detectWithIfNone(
+            Predicate2<? super T, ? super P> predicate,
+            P parameter,
+            Function0<? extends T> function)
+    {
+        T result = ListIterate.detectWith(this, predicate, parameter);
+        return result == null ? function.value() : result;
+    }
```

Comment:
I don't see how this override of detectIfWithNone is better than the implementation it overrides. It goes through a different path but still winds up calling detect(). And it's one more place with a null check. Could we just skip the override of detectIfNone and detectWithIfNone?